test/test_network.h

```diff
@@ -436,15 +436,17 @@ TEST(network, gradient_check3) {  // mixture - mse
 }
 
 TEST(network, gradient_check4) {  // sigmoid - cross-entropy
-  typedef cross_entropy loss_func;
-  typedef sigmoid activation;
-  typedef network<sequential> network;
+  using loss_func        = cross_entropy;
+  using activation_layer = sigmoid_layer;
+  using network          = network < sequential.;
 
   network nn;
-  nn << fully_connected_layer<activation>(10, 14 * 14 * 3)
-     << convolutional_layer<activation>(14, 14, 5, 3, 6)
-     << average_pooling_layer<activation>(10, 10, 6, 2)
-     << fully_connected_layer<activation>(5 * 5 * 6, 3);
+  nn << fully_connected_layer(10, 14 * 14 * 3) << activation_layer(14 * 14 * 3)
+     << convolutional_layer<identity>(14, 14, 5, 3, 6)
```

Comment:
Yes, next PR for sure. Just wanted to try out dropping from one layer in this PR.